### PR TITLE
Added new data source `azurerm_mariadb_server`

### LINKED ISSUE #5088: Support for "azurerm_mariadb_server"
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
There should be a similar data source to `azurerm_sql_server` for `azurerm_mariadb_server`.

### New or Affected Resource(s)

New data source:
* azurerm_mariadb_server

### Potential Terraform Configuration

```hcl
data "azurerm_mariadb_server" "db_server" {
  name                = "mariadb-server"
  resource_group_name = "example"
}
```

### PR DESCRIPTION
**Pull request by tracypholmes**
Fixes #5088.
image

### CODE REVIEW

File: azurerm/internal/services/mariadb/tests/data_source_arm_mariadb_server_test.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourceAzureRMMariaDbServer_basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
```
Comment (katbyte) [inline azurerm/internal/services/mariadb/tests/data_source_arm_mariadb_server_test.go]: could we use 
```suggestion
  name     = "acctestRG-maria-%d"
```

File: azurerm/internal/services/mariadb/tests/data_source_arm_mariadb_server_test.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+					resource.TestCheckResourceAttr(data.ResourceName, "administrator_login", "acctestun"),
+					resource.TestCheckResourceAttr(data.ResourceName, "version", "10.2"),
+					resource.TestCheckResourceAttr(data.ResourceName, "ssl_enforcement", "Enabled"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourceAzureRMMariaDbServer_basicOldSku(data acceptance.TestData) string {
```
Comment (katbyte) [inline azurerm/internal/services/mariadb/tests/data_source_arm_mariadb_server_test.go]: We don't need more then 1 datasource test usually

File: website/docs/d/mariadb_server.html.markdown [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+The following attributes are exported:
+
+* `id` - The ID of the MariaDB Server.
+
+* `fqdn` - The FQDN of the MariaDB Server.
+
+* `location` - The Azure location where the resource exists.
+
+* `sku_name` - Specifies the SKU Name for this MariaDB Server. The name of the SKU, follows the `tier` + `family` + `cores` pattern (e.g. `B_Gen4_1`, `GP_Gen5_8`). For more information see the [product documentation](https://docs.microsoft.com/en-us/rest/api/mariadb/servers/create#sku).
```
Comment (katbyte) [inline website/docs/d/mariadb_server.html.markdown]: As this is a data source we don't need
```suggestion
* `sku_name` - Specifies the SKU Name for this MariaDB Server.
```

File: website/docs/d/mariadb_server.html.markdown [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+* `storage_profile` - A `storage_profile` block as defined below.
+
+* `administrator_login` - The Administrator Login for the MariaDB Server.
+
+* `administrator_login_password` - The Password associated with the `administrator_login` for the MariaDB Server.
+
+* `version` - Specifies the version of MariaDB being used. Possible values are `10.2` and `10.3`.
```
Comment (katbyte) [inline website/docs/d/mariadb_server.html.markdown]: And we use data source wording
```suggestion
* `version` - The version of MariaDB being used.
```

File: azurerm/internal/services/mariadb/tests/data_source_arm_mariadb_server_test.go (hunk 1 above)
Comment (tracypholmes) [reply azurerm/internal/services/mariadb/tests/data_source_arm_mariadb_server_test.go]: It won't let me commit your suggestion on this one (greyed out?), but I made the change.

File: azurerm/internal/services/mariadb/data_source_mariadb_server.go [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+	if sku := resp.Sku; sku != nil {
+		d.Set("sku_name", sku.Name)
+	}
+
+	if properties := resp.ServerProperties; properties != nil {
+		d.Set("administrator_login", properties.AdministratorLogin)
+		d.Set("version", string(properties.Version))
+		d.Set("ssl_enforcement", string(properties.SslEnforcement))
+		// Computed
```
Comment (katbyte) [inline azurerm/internal/services/mariadb/data_source_mariadb_server.go]:
```suggestion
```

File: azurerm/internal/services/mariadb/data_source_mariadb_server.go [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+		d.Set("fqdn", properties.FullyQualifiedDomainName)
+
+		if err := d.Set("storage_profile", flattenMariaDbStorageProfile(properties.StorageProfile)); err != nil {
+			return fmt.Errorf("Error setting `storage_profile`: %+v", err)
+		}
+	}
+
+	if err := d.Set("sku", flattenMariaDbServerSku(resp.Sku)); err != nil {
```
Comment (katbyte) [inline azurerm/internal/services/mariadb/data_source_mariadb_server.go]: This doesn't exist in schema?

Comment (tracypholmes) [reply azurerm/internal/services/mariadb/data_source_mariadb_server.go]: I looked right over that. Thanks for catching it!

**Pull request merged**